### PR TITLE
feat: log swap errors to sentry

### PR DESCRIPTION
**Pull request by just-toby**
## Description

logs swap failures to sentry for more data gathering

_Linear ticket:_ https://linear.app/uniswap/issue/WEB-2210/investigate-cx-reported-swap-failures
_Slack thread:_ https://uniswapteam.slack.com/archives/C03ESADBQPK/p1685973352773609

### CODE REVIEW

File: src/pages/Swap/index.tsx [hunk 1]
```diff
@@ -440,6 +441,9 @@ export function Swap({
         })
       })
       .catch((error) => {
+        if (!didUserReject(error)) {
```
[the remark this answers is not in the document]
Comment (just-toby) [reply src/pages/Swap/index.tsx:444]: we do handle this differently in the modal UI, this is just making sure we don't log intentional rejects to sentry

[the remark this answers is not in the document]
Comment (vm) [reply src/pages/Swap/index.tsx:444]: ah okay perfect!

**Comment by vm** [comment]
Can you attempt to test this? Any errors logged locally should send to the development interface project in Sentry. You may not need to stringify the trade object for example

**Review by zzmp — APPROVED** [review]
is this the correct way to log failures? this will come through as a sentry exception that could trigger pagerduty - is that intentional?

**Comment by just-toby** [comment]
> is this the correct way to log failures? this will come through as a sentry exception that could trigger pagerduty - is that intentional?

yes that's what we want to start - it seems easy enough to silence on the sentry side

**Review by vm — APPROVED** [review]

**Pull request merged**